Modules/xSharePoint/DSCResources/MSFT_xSPFeature/MSFT_xSPFeature.psm1

```diff
@@ -47,23 +49,55 @@ function Set-TargetResource
         [parameter(Mandatory = $true)]  [System.String] $Url,
         [parameter(Mandatory = $true)]  [ValidateSet("Farm","WebApplication","Site","Web")] [System.String] $FeatureScope,
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount,
-        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure
+        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure,
+        [parameter(Mandatory = $false)] [System.String] $Version
     )
 
     $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
         $params = $args[0]
         
 
-        $runParams = @{}
-        $runParams.Add("Identity", $params.Name)
+        $runParams = @{ 
+            Identity = $params.Name 
+            Verbose = $false
+        }
+
         if ($params.FeatureScope -ne "Farm") {
             $runParams.Add("Url", $params.Url)
         }
 
+        # Parameters for get
+        $checkParams = @{ 
+            Identity = $params.Name 
+            Verbose = $false    
+        }
+        
+        if ($params.FeatureScope -eq "Farm") 
+        {
+            $checkParams.Add($params.FeatureScope, $true)
+        } 
+        else
+        {
+            $checkParams.Add($params.FeatureScope, $params.Url)
+        }
```

Comment:
Rather than run this same logic from the get method, can you call Get-TargetResource to determine what you need here rather than duplicate the code?